app/lib/actions/katello/repository/clone_yum_metadata.rb

```diff
@@ -20,8 +20,16 @@ def plan(source_repo, target_repo)
                         target_repo,
                         :source_repository => source_repo,
                         :matching_content => output[:matching_content])
+
+            plan_self(:source_checksum_type => source_repo.saved_checksum_type, :target_repo_id => target_repo.id) unless source_repo.saved_checksum_type == target_repo.saved_checksum_type
           end
         end
+
+        def finalize
+          repository = ::Katello::Repository.find(input[:target_repo_id])
+          source_checksum_type = input[:source_checksum_type]
+          repository.update_attributes!(saved_checksum_type: source_checksum_type) if (repository && source_checksum_type)
+        end
```

Comment:
This block is to update the checksum type on the library repo if we did get a difference between the checksum types on the new version of repo and the library repo. If we see a difference, we plan the finalize step and update the checksum_type on library repo so that next time we check for diffs, we are checking against the updated checksum_type.